es6.md

```diff
@@ -189,3 +189,35 @@ interface ImportNamespaceSpecifier {
 ```
 
 A namespace import specifier, e.g., `* as foo` in `import * as foo from "mod.js"`.
+
+## ExportDeclaration
+
+```js
+interface ExportDeclaration <: Node {
```

Comment:
I always feel like I have to sit down and think really hard about this structure because it represents so many different semantic forms. I'm always worried about screwing it up because it has so many states to consider -- many of which are even invalid in unobvious ways: 

* Having `default` be true and `source` be non-null is, currently at least, invalid
* Having `default` be true and `specifiers` be an empty list -- or a list of more than 1 -- is invalid
* Having multiple BatchSpecifiers in a specifiers list is invalid
* Having a BatchSpecifier in a list with *anything* else is invalid
* Having a BatchSpecifier with `default=true` is invalid

(there are probably more permutations...)

It would make this so much easier to reason about if we had a node-type for each of those forms (something like `ExportBatchDeclaration`, `ExportDefaultDeclaration`, `ExportNamedDeclaration`).

I don't think having zero invalid states is a goal we should shoot for in the general case, but when the number of invalid states is so high (and relatedly difficult to reason about), I think it becomes a problem.